lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1761,19 +1842,19 @@ def _get_json_from_tar_file(b_path, filename):
     return json.loads(file_contents)
 
 
-def _get_tar_file_hash(b_path, filename):
+def _get_tar_file_hash(b_path: bytes, filename: str) -> str:
     with tarfile.open(b_path, mode='r') as collection_tar:
         with _get_tar_file_member(collection_tar, filename) as (dummy, tar_obj):
             return _consume_file(tar_obj)
 
 
-def _get_file_hash(b_path, filename):  # type: (bytes, str) -> str
+def _get_file_hash(b_path: bytes, filename: str) -> str:
     filepath = os.path.join(b_path, to_bytes(filename, errors='surrogate_or_strict'))
     with open(filepath, 'rb') as fp:
         return _consume_file(fp)
 
 
-def _is_child_path(path, parent_path, link_name=None):
+def _is_child_path(path: bytes, parent_path: bytes, link_name=None) -> bool:
```

Comment:
```suggestion
def _is_child_path(path: bytes, parent_path: bytes, link_name: bytes | None = None) -> bool:
```

Since we pass in bytes, the to_bytes conversion for path, parent_path, and link_name could be removed from this function.

Reply:
True. Though since I haven't checked the entire call stack, I didn't want to make runtime changes just in case.

Thanks for catching the type, though. Sometimes, `pyrefly autotype` does not annotate every arg, just portion of them. I was mostly working on things that started failing in MyPy once types started to be checked in more places.